trek.js

```diff
@@ -0,0 +1,149 @@
+TRIP_URL = "https://trektravel.herokuapp.com/trips/"
+
+// error reporting so the user knows what went wrong
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+};
+
+const reportApiError = (error) => {
+  console.log("We've encountered the following error:", error);
+
+  let errorHtml = `<p>${error.message}</p><ul>`;
+
+  const allProblems = error.response.data.errors;
+
+ 
+  Object.keys(allProblems).forEach(key => {
```

Comment:
It's nice that you're reporting errors to the user, but you should also adapt this to handle network issues rather than just validation errors.